source/common/http/hash_policy.cc

```diff
@@ -39,14 +42,27 @@ class HeaderHashMethod : public HashMethodImplBase {
                                     const StreamInfo::FilterStateSharedPtr) const override {
     absl::optional<uint64_t> hash;
 
-    // TODO(mattklein123): Potentially hash on all headers.
     const auto header = headers.get(header_name_);
     if (!header.empty()) {
+      std::vector<std::string> rewrited_header_values;
+      std::vector<absl::string_view> header_values;
+      header_values.reserve(header.size());
       if (regex_rewrite_ != nullptr) {
-        hash = HashUtil::xxHash64(regex_rewrite_->replaceAll(header[0]->value().getStringView(),
-                                                             regex_rewrite_substitution_));
-      } else {
-        hash = HashUtil::xxHash64(header[0]->value().getStringView());
+        rewrited_header_values.reserve(header.size());
+      }
+      for (size_t i = 0; i < header.size(); i++) {
+        if (regex_rewrite_ != nullptr) {
```

Comment:
see my previous comment, which proposes a  more efficient because it only does the regex_rewrite_==null check once, rather than once per header.

Reply:
I thought it would be twice the loop for having regex_rewrite. But I see, it is more efficient for the case of not having regex_rewrite.